zookeeper-server/src/main/java/org/apache/zookeeper/server/SnapshotFormatter.java

```diff
@@ -34,79 +36,139 @@
-        if (args.length != 1) {
-            System.err.println("USAGE: SnapshotFormatter snapshot_file");
+        String snapshotFile = null;
+        boolean dumpData = false;
+        boolean dumpJson = false;
+
+        int i;
+        for (i = 0; i < args.length; i++) {
+            if (args[i].equals("-d")) {
+                dumpData = true;
+            } else if (args[i].equals("-json")) {
+                dumpJson = true;
+            } else {
+                snapshotFile = args[i];
+                i++;
+                break;
+            }
+        }
+        if (args.length != i || snapshotFile == null) {
+            System.err.println("USAGE: SnapshotFormatter [-d|-json] snapshot_file");
+            System.err.println("       -d dump the data for each znode");
+            System.err.println("       -json dump znode info in json format");
             System.exit(2);
         }
 
-        new SnapshotFormatter().run(args[0]);
+        if (dumpData && dumpJson) {
+            System.err.println("Cannot specify both data dump (-d) and json mode (-json) in same call");
+            System.exit(ExitCode.INVALID_INVOCATION.getValue());
+        }
+
+        new SnapshotFormatter().run(snapshotFile, dumpData, dumpJson);
     }
-    
-    public void run(String snapshotFileName) throws IOException {
-        InputStream is = new CheckedInputStream(
+
+    public void run(String snapshotFileName, boolean dumpData, boolean dumpJson)
+        throws IOException {
+        File snapshotFile = new File(snapshotFileName);
+        try (InputStream is = new CheckedInputStream(
                 new BufferedInputStream(new FileInputStream(snapshotFileName)),
-                new Adler32());
-        InputArchive ia = BinaryInputArchive.getArchive(is);
-        
-        FileSnap fileSnap = new FileSnap(null);
+                new Adler32())) {
+            InputArchive ia = BinaryInputArchive.getArchive(is);
+
+            FileSnap fileSnap = new FileSnap(null);
+
+            DataTree dataTree = new DataTree();
+            Map<Long, Integer> sessions = new HashMap<Long, Integer>();
 
-        DataTree dataTree = new DataTree();
-        Map<Long, Integer> sessions = new HashMap<Long, Integer>();
-        
-        fileSnap.deserialize(dataTree, sessions, ia);
+            fileSnap.deserialize(dataTree, sessions, ia);
+            long fileNameZxid = Util.getZxidFromName(snapshotFile.getName(), SNAPSHOT_FILE_PREFIX);
 
-        printDetails(dataTree, sessions);
+            if (dumpJson) {
+                printSnapshotJson(dataTree);
+            } else {
+                printDetails(dataTree, sessions, dumpData, fileNameZxid);
+            }
+        }
     }
 
-    private void printDetails(DataTree dataTree, Map<Long, Integer> sessions) {
-        printZnodeDetails(dataTree);
+    private void printDetails(
+        DataTree dataTree, Map<Long, Integer> sessions,
+        boolean dumpData, long fileNameZxid
+    ) {
```

Comment:
`) {`  this formatting does not look good. format it

Reply:
did the formatting. could you please review it.